pwiz_tools/Skyline/Controls/Lists/ListGridForm.cs

```diff
@@ -67,5 +67,18 @@ public override DataGridId DataGridId
                 return new DataGridId(DataGridType.LIST, ListName);
             }
         }
+
+        protected override string GetPersistentString()
+        {
+            return base.GetPersistentString() + '|' + ListName;
+        }
+
+        public static string GetListName(string persistentString)
+        {
+            string[] values = persistentString.Split('|');
```

Comment:
There's a helper class "PersistentString" which takes care of escaping in case parts of the string contain "|" characters.